server/container_create_linux.go

```diff
@@ -548,36 +505,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 
 		if !isInCRIMounts("/sys", containerConfig.GetMounts()) {
 			specgen.RemoveMount("/sys")
-			specgen.RemoveMount("/sys/fs/cgroup")
-			sysMnt := rspec.Mount{
-				Destination: "/sys",
-				Type:        "sysfs",
-				Source:      "sysfs",
-				Options:     []string{"nosuid", "noexec", "nodev", "ro"},
-			}
-			specgen.AddMount(sysMnt)
+			ctr.SpecAddMount("/sys", "sysfs", "sysfs", "/sys/fs/cgroup", []string{"nosuid", "noexec", "nodev", "ro"})
```

Comment:
After reviewing the whole commit, it seems this change is hurting readability. While previously we had code which clearly shows what's what:
```go
specgen.RemoveMount("/sys/fs/cgroup")
sysMnt := rspec.Mount{
￼				Destination: "/sys",
￼				Type:        "sysfs",
￼				Source:      "sysfs",
...
```

now we have 
```go
ctr.SpecAddMount("/sys", "sysfs", "sysfs", "/sys/fs/cgroup", ...
```
which is somewhat harder to read.

I could be OK with the increased readability in case the change would add some benefit, but I don't see it.

That's just IMHO though, feel free to ignore.

Reply:
what if we passed in an rspec.Mount object?
Then SpecAddMount would RemoveMount(m.Destination), then AddMount(m)

The idea behind the change is to move the work of translating between cri->oci from server/container_create_linux to pkg/container. Eventually, more will be moved and the purpose will be more clear, and I believe readability will be improved

Reply:
passing in an rspec.Mount object now!